pkg/stdcopy/stdcopy.go

```diff
@@ -155,11 +160,18 @@ func StdCopy(dstout, dsterr io.Writer, src io.Reader) (written int64, err error)
 			}
 		}
 
+		// we might have an error from the daemon mixed up in our multiplexed http
+		// stream. if we do, return it.
+		if out == nil {
+			return written, fmt.Errorf("got error from daemon: %v", string(buf[stdWriterPrefixLen:frameSize+stdWriterPrefixLen]))
```

Comment:
Minor: `%s` seems like the right format specifier to use here because it's more explicit and the argument is obviously a string. It shouldn't matter though.

Is returning an error right away the correct behavior here? Based on the uses of `Systemerr` you're adding in this PR, it looks fine. But I wonder if it's better to store the errors and return them after reaching the end of the input.